src/main/java/io/github/railroad/menu/TopMenu.java

```diff
@@ -0,0 +1,238 @@
+package io.github.railroad.menu;
+
+import javafx.scene.control.Menu;
+import javafx.scene.control.MenuBar;
+import javafx.scene.control.SeparatorMenuItem;
+
+import static io.github.railroad.lang.LangManager.ENGLISH;
+import static io.github.railroad.lang.LangManager.getLocalization;
+import static io.github.railroad.utility.Components.MenuItemFactory.makeMenuItem;
+import static io.github.railroad.utility.Components.createImage;
+import static io.github.railroad.utility.FileChoosers.createNewGenericFile;
+import static io.github.railroad.utility.FileChoosers.createNewJavaFile;
+import static io.github.railroad.utility.Templates.JavaTemplate.*;
+import static io.github.railroad.utility.Terminal.openTerminal;
+
+/**
+ * Creates all of the items of the top menu.
+ *
+ * @author TheOnlyTails
+ */
+public final class TopMenu extends MenuBar {
+    /**
+     * Creates the top menu and adds all of the submenus.
+     *
+     * @author TheOnlyTails
+     */
+    public void createMenu() {
+        final var fileMenu = new Menu(getLocalization("menu.file", ENGLISH));
+        createFileMenu(fileMenu);
+
+        final var editMenu = new Menu(getLocalization("menu.edit", ENGLISH));
+        createEditMenu(editMenu);
+
+        final var searchMenu = new Menu(getLocalization("menu.search", ENGLISH));
+        createSearchMenu(searchMenu);
+
+        final var runMenu = new Menu(getLocalization("menu.run", ENGLISH));
+        final var viewMenu = new Menu(getLocalization("menu.view", ENGLISH));
+        createViewMenu(viewMenu);
+
+        final var helpMenu = new Menu(getLocalization("menu.help", ENGLISH));
+        getMenus().addAll(fileMenu, editMenu, searchMenu, runMenu, viewMenu, helpMenu);
+    }
+
+    /**
+     * Creates the file submenu and all of its submenus.
+     *
+     * @param fileMenu the object that contains the file submenu.
+     * @author TheOnlyTails
+     */
+    public void createFileMenu(Menu fileMenu) {
+        createFileNewMenu(fileMenu);
+        createFileGenerateMenu(fileMenu);
+        fileMenu.getItems().add(new SeparatorMenuItem());
+
+        final var open = makeMenuItem(getLocalization("menu.file.open", ENGLISH))
+                .graphic(createImage("/assets/img/java_project.png")).get();
```

Comment:
Would it not be better to store these images somewhere, so that you're not creating the same thing over and over and over again?

Reply:
Yeah, I'll so it tomorrow.